app/validators/fee/fixed_fee_validator.rb

```diff
@@ -11,6 +11,16 @@ def self.fields
 
   private
 
+  def validate_claim
+    super
+    if @record.claim
```

Comment:
Would be nicer as `if @record.claim && @record.claim.final? && !@record.claim.case_type.is_fixed_fee?` to reduce nesting and complexity, I think.